### ISSUE TITLE
I install latest microG have no system spoofs signature in Self-Check

### ISSUE THREAD

**Issue by devincer** [microg/GmsCore#2667]
paly services (Gmscore) has correct signature is not enable.
it tips :
Either the installed Play Services (GmsCore)
is not compatible or signature spoofing is not
active for it. Please check the documentation on
which applications and ROMs are compatible.

my phone : huawei mate30E pro 5G  
LIO_AN00m

My system is huawei harmony os 3.0, I installed the version of microG v0.3.2.240913, gave me this prompt.
I upgraded my operating system to harmonyOS 4.0.0.113 and installed the latest microG v0.3.4.240913, but still no system spoofs signature was displayed and gave me this tip. The situation remains the same.

Apps such as youtube and chatgpt cannot be used if they are not properly signed.

I don't know how to complete this system spoofs signature.
Does it look like a system hardware problem? My phone is not in microG and GMS support?
Or is it the system software? My play service version is wrong or my microG is not working properly?
What do I need to do to enable spoofs signature and then where to view the adaptation document?
Or which other tool is needed to complete the fake signature?

**Comment by ale5000-git** [comment]
The version 0.3.4 doesn't exist for Huawei.
See here: #2624

**Comment by MugoSquero** [comment]
If you are rooted, you can try [LSposed](https://github.com/JingMatrix/LSPosed) module [FakeGApps](https://github.com/whew-inc/FakeGApps) to spoof the siganture of MicroG apps to correct one. Don't know anything about Huawei phones though, a **-hw** release might be necessary.

**Comment by ale5000-git** [comment]
The **-hw** version was created mainly to get signature spoofing on Huawei.

**Comment by devincer** [comment]
> The **-hw** version was created mainly to get signature spoofing on Huawei.

i installed microG v0.3.2 -hw still show me this prompt. so i decide upgrade os system and microG .
the latest microG v0.3.2 **-hw** not working .seems it has no signature.
:(

**Comment by ale5000-git** [comment]
The latest version of **hw** _microG Services_ is currently 0.3.3.1.240913-hw.
You need both microG Services (com.google.android.gms-hw) and microG Companion (com.android.vending-hw).

They can be downloaded here: https://microg.org/download.html (the two links at the bottom)

**Comment by devincer** [comment]
> The latest version of **hw** _microG Services_ is currently 0.3.3.1.240913-hw. You need both microG Services (com.google.android.gms-hw) and microG Companion (com.android.vending-hw).
> 
> They can be downloaded here: https://microg.org/download.html (the two links at the bottom)

OK,thanks for all your help.
it works successful.

**Comment by ivan-volnov** [comment]
Current the latest version (0.3.6). Problem exists. Downgraded to 0.3.3 helped.

**Comment by IOsetting** [comment]
I installed the latest microG on Huawei tablet and met the same problem.

Version info:

* HarmonyOS 3.1.0.175 
* microG Services v0.3.6.244735-hw
* microG Companion v0.3.6.40226-hw
* microG Services Framework Proxy v0.1.0

**Comment by ale5000-git** [comment]
The **hw** get signature spoofing from the Huawei ROM, so it probably need to be fixed by Huawei.

**Comment by UjuiUjuMandan** [comment]
> The **hw** version get signature spoofing from the Huawei ROM, so it probably need to be fixed by Huawei.

@ale5000-git Hi, can you add HarmonyOS to the [wiki page](https://github.com/microg/GmsCore/wiki/Signature-Spoofing) with some explanation about their implementation on this?

**Comment by UjuiUjuMandan** [comment]
Great word choosing...

https://github.com/microg/GmsCore/blob/e2b4d38f45689f6384535cadec0c1d489173d1ae/fake-signature/src/huawei/aidl/com/huawei/signature/diff/ISignatureService.aidl#L9-L14

**Comment by Kzet98** [comment]
I'm new to this, but I may be running into the same issue. I have installed microG Services (v0.3.13.2.250932) to work with YouTube Revanced (v20.40.45) on my Moto G05, and since I wasn't receiving notifications push and YTRV kept disconnecting in background despite multiple precautions for battery usage, last thing I did was checking if signature spoof was missing. Admittedly, I'm not seeing it anywhere in the app, but I dunno if it's by design (again, huge newbie). Any help for getting my notifications to work is much appreciated.
